commit.go

```diff
@@ -137,6 +137,30 @@ func (c *Commit) History() ([]*Commit, error) {
 	return commits, err
 }
 
+// Encode transforms a Commit into a core.Object.
+func (b *Commit) Encode(o core.Object) error {
+	o.SetType(core.CommitObject)
+
+	w, err := o.Writer()
+	if err != nil {
+		return err
+	}
+	w.Write([]byte("tree "))
+	w.Write([]byte(b.tree.String()))
```

Comment:
should we change this to:
```
fmt.Fprintf(w, "tree %s", b.tree.String())
```

I have not benchmark this against the proposed solution, but if it is not too slow, it is definitely more idiomatic and clear, especially if you do the same on lines 154-...